internal/app/component/component_handler_test.go

```diff
@@ -254,4 +306,57 @@ var _ = Describe("When deleting Component", Label("app", "DeleteComponent"), fun
 		Expect(err).To(BeNil(), "no error should be thrown")
 		Expect(components.Elements).To(BeEmpty(), "no error should be thrown")
 	})
+
+	Context("when handling an DeleteComponentEvent", func() {
+		BeforeEach(func() {
+			db.On("GetDefaultIssuePriority").Return(int64(100))
+			db.On("GetDefaultRepositoryName").Return("nvd")
+		})
+
+		Context("when new component is deleted", func() {
+			It("should delete tuples related to that component in openfga", func() {
+				// Test OnComponentDeleteAuthz against all possible relations
+				compFake := test.NewFakeComponentEntity()
+				deleteEvent := &c.DeleteComponentEvent{
+					ComponentID: compFake.Id,
+				}
+				objectId := openfga.ObjectId(strconv.FormatInt(deleteEvent.ComponentID, 10))
+				relations := []openfga.RelationInput{
+					{ // role - component: a role is assigned to the component
```

Comment:
The relationship creation looks confusing. If it's correct, we might need to rename the attributes in `RelationInput`. 
```
UserType: "component_version"
UserId: "cvID"
```